.ci/Jenkinsfile

```diff
@@ -131,6 +132,43 @@ pipeline {
         pushDockerImage()
       }
     }
+    stage('v2: Publish packages') {
+      steps {
+        script {
+          cleanup()
+          dir("${BASE_DIR}") {
+            dir("packages") {
+              def packageRevision = [:]
+              findFiles()?.findAll{ !it.name.endsWith('@tmp') }?.collect{ it.name }?.sort()?.each {
+                def packageName = it
+                dir("${packageName}") {
+                  findFiles()?.findAll{ !it.name.endsWith('@tmp') }?.collect{ it.name }?.sort()?.each {
+                    def packageVersion = it
+                    // Check if the package revision is published
+                    def response = httpRequest(httpMode: "HEAD", url: "https://package-storage.elastic.co/artifacts/packages/${packageName}-${packageVersion}.zip",
```

Comment:
Hi @v1v, I noticed that we're using our homebuild function `httpRequest`, not the [plugin](https://www.jenkins.io/doc/pipeline/steps/http_request/). I was wondering if we can add an extra parameter `validResponseCodes` there. Otherwise, I will have to catch exceptions :) WDYT?

Reply:
As long as it keeps the backward-compatibility, I'd say go for it, there is also an option to create a new step which enriches the `httpRequest` step, for instance -> `httpRequestExt()` if it also helps

Reply:
Thanks for jumping in! I found that there is also a property `response_code_only`. I will try to use it as it might fit here better.